### PR TITLE
Faster pubcloud registration

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/143085/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 818f4acc3658789028c18345e057b8fb12fcf75e into 0e4c24181771ff4d66ee13b0fee6c438d16831d2)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/143947/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 4feeda10c380d26b60c5c356836b7abd52757131 into 0e4c24181771ff4d66ee13b0fee6c438d16831d2)

File: lib/suse/connect/client.rb [hunk 1]
```diff
@@ -26,19 +26,19 @@ def register!
         announce_or_update
         product = @config.product || Zypper.base_product
 
-        register_product(product, @config.product ? true : false)
+        register_product(product, @config.product ? true : false, !@config.no_zypper_refs)
```
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply lib/suse/connect/client.rb]: Seeing that `register_product` already reads the config -- yeah, that probably makes sense. I wouldn't then need to have the parameter for `register_product_tree` at all.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/144228/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 36794205b860b7b39febf041ee57481ed5647b1d into 0e4c24181771ff4d66ee13b0fee6c438d16831d2)

File: lib/suse/connect/client.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply lib/suse/connect/client.rb]: This is done, the code should be a bit simpler now.

File: spec/connect/client_spec.rb [hunk 2]
```diff
@@ -343,8 +343,8 @@
       before { recommended_2.available = false }
 
       it 'does not register the unavailable extension nor its children' do
-        expect(subject).not_to receive(:register_product).with(recommended_2)
-        expect(subject).not_to receive(:register_product).with(recommended_2_2)
+        expect(subject).not_to receive(:register_product).with(recommended_2, any_args)
```
Comment (felixsch) [inline spec/connect/client_spec.rb]: the `any_args` should not be needed anymore right?

File: lib/suse/connect/client.rb [hunk 3]
```diff
@@ -34,11 +34,11 @@ def register!
         log.info 'Successfully registered system.'
       end
 
-      # Activate the product, add the service and install the relase package
+      # Activate the product, add the service and install the release package
       def register_product(product, install_release_package = true)
         service = activate_product(product, @config.email)
 
-        System.add_service(service)
+        System.add_service(service, !@config.no_zypper_refs)
```
Comment (felixsch) [inline lib/suse/connect/client.rb]: In general I dislike this double negation: Its harder to read than `zypper_refresh = false|true` especially in the spec files. But anyway in this case its fine.

Comment (felixsch) [reply lib/suse/connect/client.rb]: I know but default can be also `true`. Anyway I have no problem with this. Just more like a comment without an request to change :P

File: spec/connect/client_spec.rb (hunk 2 above)
Comment (felixsch) [reply spec/connect/client_spec.rb]: Yes but why adding it if there is no need for it. Also it adds the possibility to execute `register_product` with `any_args`.

File: lib/suse/connect/client.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply lib/suse/connect/client.rb]: True. Fixed.

File: spec/connect/client_spec.rb (hunk 2 above)
Comment (ikapelyukhin) [reply spec/connect/client_spec.rb]: True. Fixed.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/144446/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 1e0835e85af6fe73ae4f2d720b8a2d7dbc17267f into 0e4c24181771ff4d66ee13b0fee6c438d16831d2)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/144485/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged bb27ec27e6386255c81cccf78d7584ade708795d into 0e4c24181771ff4d66ee13b0fee6c438d16831d2)

**Pull request merged**